File: backend/blueprints/spa_api/service_layers/chart_data.py
from typing import List, Optional


class ChartDataPoint:
    def __init__(self, name: str, value: float, average: Optional[float] = None):
        self.name = name
        self.value = value
        if average is not None:
            self.average = average


class ChartData:
    def __init__(self, title: str, chart_data_points: List[ChartDataPoint]):
        self.title = title
        self.chartDataPoints = [chart_data_point.__dict__ for chart_data_point in chart_data_points]

File: webapp/src/Models/ChartData.ts
export interface ChartDataPoint {
    name: string
    value: number
    average?: number
}

export interface ChartDataResponse {
    title: string
    chartDataPoints: ChartDataPoint[]
}

export interface StatDataPoint extends ChartDataPoint {
    isOrange: boolean
}

export interface BasicStat extends ChartDataResponse {
    chartDataPoints: StatDataPoint[]
    type: "radar" | "bar" | "pie"
    subcategory: StatsSubcategory
}

export enum PlayerStatsSubcategory {
    MAIN_STATS = "Main Stats",
    POSITIONING = "Positioning",
    HITS = "Hits",
    BALL = "Ball",
    BOOSTS = "Boosts",
    PLAYSTYLES = "Playstyles",
    POSSESION = "Possession",
    EFFICIENCY = "Efficiency",
    TEAM_POSITIONING = "Team Positioning"
}

export enum TeamStatsSubcategory {
    POSITIONING = "Positioning",
    CENTER_OF_MASS = "Center of Mass"
}

export enum HeatmapSubcategory {
    POSITIONING = "Positioning",
    // HITS = "Hits",
    BOOST = "Boost",
    BOOST_SPEED = "Boost Speed",
    SLOW_SPEED = "Slow Speed"
}

export type StatsSubcategory = PlayerStatsSubcategory | TeamStatsSubcategory
